### PR TITLE
[match] propagate keychain when installing wwdr certificates

### PR DESCRIPTION
**Pull request by rabbitinspace**
### Checklist

- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I see several green `ci/circleci` builds in the "All checks have passed" section of my PR ([connect CircleCI to GitHub](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-) if not)
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.
- [x] I've added or updated relevant unit tests.

### Motivation and Context

After updating from fastlane 2.202.0 to 2.216.0 `fastlane match -s $KEYCHAIN -p $PASSWORD` started to fail for me with the following error:

```shell
[03:54:46]: 🔓  Successfully decrypted certificates repo
[03:54:46]: Installing certificate...
[03:54:46]: $ security find-certificate -a -c 'Apple Worldwide Developer Relations' -p /Users/ec2-user/Library/Keychains/login.keychain-db
[!] Could not install WWDR certificate
🚨 Error: The command exited with status 1
``` 

I import wwdr certs manually after creating a keychain, so something is broken here. The weird thing I found is `security find-certificate` is invoked for the login keychain despite me specifying a separate one. After digging a bit, I figured the `install_missing_wwdr_certificates` method doesn't take into account the `-s` option.

### Description

This PR makes the `-s` option to be propagated to the `install_missing_wwdr_certificates` method instead of unconditionally using `wwdr_keychain`. I found [this issue](https://github.com/fastlane/fastlane/issues/20960) which supposed to be fixed, but I guess this edge case isn't handled.

Also, I hope I fixed the tests the right way.

### Testing Steps

```shell
KEYCHAIN_NAME=certs
KEYCHAIN_PATH="${HOME}/Library/Keychains/${KEYCHAIN_NAME}.keychain-db"
KEYCHAIN_PASS=12345HACKER

# create a new keychain first

if [[ -f "${KEYCHAIN_PATH}" ]]; then
  fastlane run delete_keychain keychain_path:"${KEYCHAIN_PATH}"
fi
fastlane run create_keychain \
  password:"${KEYCHAIN_PASS}" \
  path:"${KEYCHAIN_PATH}" \
  unlock:true \
  timeout:1800

# import all wwdr certs

certs=( AppleWWDRCA.cer AppleWWDRCAG2.cer AppleWWDRCAG3.cer AppleWWDRCAG4.cer AppleWWDRCAG5.cer AppleWWDRCAG6.cer )
for cert in "${certs[@]}"; do
  fastlane run import_certificate \
    certificate_path:"${cert}" \
    keychain_password:"${KEYCHAIN_PASS}" \
    keychain_path:"${KEYCHAIN_PATH}"
done

# run fastlane match to import certs and profiles

fastlane match development -s "${KEYCHAIN_NAME}" -p "${KEYCHAIN_PASS}"
```

### CODE REVIEW

File: fastlane_core/lib/fastlane_core/cert_checker.rb [hunk 1]
```diff
@@ -111,12 +112,12 @@ def self.list_available_developer_id_installer(in_keychain: nil)
       `#{commands.join(' ')}`
     end
 
-    def self.installed_wwdr_certificates
+    def self.installed_wwdr_certificates(keychain)
```
Comment (lacostej) [inline fastlane_core/lib/fastlane_core/cert_checker.rb:115]: from an API compatibility perspective, I wonder if we should change this to

```
def self.installed_wwdr_certificates(keychain: nil)
   keychain ||= wwdr_keychain
```

Same for other methods.

WDYT?

Comment (rabbitinspace) [reply fastlane_core/lib/fastlane_core/cert_checker.rb:115]: do you mean compatibility with other fastlane code which uses this class or users who might use some of its methods?
the first one should already be sorted as (iirc) I didn't change signatures for used methods
as for the second one, there are no clear public/private separation to me so I assumed those methods are implementation detail as they doesn't look like a public api to me
but happy to use your suggestion if you think it would be the right thing to do

Comment (lacostej) [reply fastlane_core/lib/fastlane_core/cert_checker.rb:115]: @rabbitinspace I meant compatibility with potential external code. 

We unfortunately have a grey way of defining API compatibility in fastlane. I know some users are sometimes using parts of the tools programmatically and anything that isn't made `private` can be used. 

So maybe

```ruby
keychain ||= wwdr_keychain # backwards compatibility
```

would make sense.

I'll ask the team if we can agree on what the clarify what the project public API is.

Comment (rabbitinspace) [reply fastlane_core/lib/fastlane_core/cert_checker.rb]: thanks! addressed in 6ad8b71bd

**Comment by rabbitinspace** [comment]
conflicts resolved, the CI failure is possibly due to https://github.com/fastlane/fastlane/issues/21663

**Review by lacostej — APPROVED** [review]
LGTM. Anyone else?

**Review by rogerluan — APPROVED** [review]
💪 LGTM! Simple and safe. Thanks for your contribution @rabbitinspace ! 🚀

**Comment by DuMaM** [comment]
Yay no more changing default keychain :D thanks man
